src/backend/src/services/users.services.ts

```diff
@@ -0,0 +1,142 @@
+import { Role, User_Settings } from '@prisma/client';
+import { OAuth2Client } from 'google-auth-library/build/src/auth/oauth2client';
+import { AuthenticatedUser, User } from 'shared';
+import { authUserQueryArgs } from '../prisma-query-args/auth-user.query-args';
+import prisma from '../prisma/prisma';
+import { authenticatedUserTransformer } from '../transformers/auth-user.transformer';
+import { userTransformer } from '../transformers/user.transformer';
+import { AccessDeniedException, NotFoundException } from '../utils/errors.utils';
+import { rankUserRole } from '../utils/users.utils';
+
+export default class UsersService {
+  static async getAllUsers(): Promise<User[]> {
+    const users = await prisma.user.findMany();
+    users.sort((a, b) => a.firstName.localeCompare(b.firstName));
+
+    return users.map(userTransformer);
+  }
+
+  static async getSingleUser(userId: number): Promise<User> {
+    const requestedUser = await prisma.user.findUnique({ where: { userId } });
+    if (!requestedUser) throw new NotFoundException('User', userId);
+
+    return userTransformer(requestedUser);
+  }
+
+  static async getUserSettings(userId: number): Promise<User_Settings> {
+    const requestedUser = await prisma.user.findUnique({ where: { userId } });
+
+    if (!requestedUser) throw new NotFoundException('User', userId);
+
+    const settings = await prisma.user_Settings.upsert({
+      where: { userId },
+      update: {},
+      create: { userId }
+    });
+
+    if (!settings) throw new NotFoundException('User Settings', userId);
+
+    return settings;
+  }
+
+  static async updateUserSettings(userId: number, defaultTheme: any, slackId: string): Promise<User_Settings> {
+    const user = await prisma.user.findUnique({ where: { userId } });
```

Comment:
this one should use the getCurrentUser thing because we don't want just anyone to be able to edit anyone's settings